### PR TITLE
Finish upgrades

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

**Comment by ace-n** [comment]
Is there a reason `codecov` is spitting out weird results?

**Comment by jmdobry** [comment]
Probably because the first time Travis built this push it attempted to run the Cloud SQL tests but the proxy wasn't running and the tests crashed and burned.

**Comment by jmdobry** [comment]
This is ready for review

File: optional-container-engine/test/background.test.js [hunk 1]
```diff
@@ -13,415 +13,386 @@
...
+    error: sinon.stub()
+  };
+  // Load background.js with provided mocks
+  background = proxyquire(`../lib/background`, {
+    '@google-cloud/pubsub': mocks.Pubsub,
+    '../config': mocks.config,
+    './logging': mocks.logging
+  });
+
+  t.truthy(
```
Comment (ace-n) [inline optional-container-engine/test/background.test.js]: Why `truthy` and not `true`?

File: optional-container-engine/test/background.test.js [hunk 2]
```diff
@@ -13,415 +13,386 @@
...
+    mocks.Pubsub.calledOnce,
+    `Pubsub() should have been called once`
```
Comment (ace-n) [inline optional-container-engine/test/background.test.js]: Do we need these individual explanations? (They're somewhat verbose)

File: optional-container-engine/test/background.test.js [hunk 3]
```diff
@@ -13,415 +13,386 @@
...
+  t.is(
+    mocks.logging.info.callCount,
+    0,
+    `logging.info() should NOT have been called`
+  );
+  t.truthy(
+    mocks.logging.error.calledOnce,
+    `logging.error() should have been called`
+  );
+  t
```
Comment (ace-n) [inline optional-container-engine/test/background.test.js]: Typo? (If so, make sure there aren't any further instances of this throughout your PR.)

File: optional-container-engine/test/background.test.js [hunk 4]
```diff
@@ -13,415 +13,386 @@
...
+  t.is(
+    mocks.pubsub.topic.callCount,
+    0,
+    `pubsub.topic() should NOT have been called`
+  );
+  t.truthy(
+    mocks.topic.publish,
+    `topic.publish() should have been called once`
+  );
+  t
```
Comment (ace-n) [inline optional-container-engine/test/background.test.js]: (Repeat of [possible] typo mentioned above.)

File: scripts/unify [hunk 5]
```diff
@@ -1,36 +1,59 @@
...
+  directories.forEach((dir) => {
+    pkgJson = JSON.parse(fs.readFileSync(path.join(directory, dir, `package.json`)));
+    Object.assign(pkgSet, pkgJson.dependencies);
+    Object.assign(pkgDevSet, pkgJson.devDependencies);
+  });
+}
+
+getDeps(PROJECT_ROOT);
+
+console.log(JSON.stringify(pkgSet, null, 2));
```
Comment (ace-n) [inline scripts/unify]: Is this debug code, or is it supposed to be here? (If it is supposed to be here, perhaps we should include a message explaining what this JSON is - e.g. `Unified package.json contents`)

File: scripts/unify [hunk 6]
```diff
@@ -1,36 +1,59 @@
...
+console.log(JSON.stringify(pkgDevSet, null, 2));
+
+// Update root-level package.json (by shelling to npm)
+const spawnSync = require('child_process').spawnSync;
+let args = [`add`].concat(Object.keys(pkgSet).map(pkg => `${pkg}@${pkgSet[pkg]}`));
+console.log(`yarn ${args.join(' ')}`);
```
Comment (ace-n) [inline scripts/unify]: Ditto my comment above.

**Pull request merged**